Add note to FAQ about cache directory

mschristiansen opened the pull request:
Just had a problem trying to use my own package set and it took me a while before I had it narrowed down and starter looking for the cache dir. Hope this note in the README can help someone else.

paf31 commented on README.md:
```diff
@@ -124,3 +124,11 @@ Not right now. We might add this feature in future, but for now, consider either
+
+### Why are my changes not updated in my package set?
+
+Package sets are cached in `.psc-package`. If you are changing a
```
When you say changing a package set, do you mean manually? Could you please clarify that? Also, could you please move the command into its own block-level element?


Thanks!

paf31 replied:
Also, maybe point out that we cache based on Git tag names?

mschristiansen committed:
Add note to FAQ about cache directory

mschristiansen commented:
I've added a more precise explanation of how the cache works and moved code to into separate block. What do you think about having a flag on the update command to pull a package set without using the cache? E.g. `psc-package update --force` will clear cache and fetch everything anew.

paf31 commented on README.md:
```diff
@@ -124,3 +124,15 @@ Not right now. We might add this feature in future, but for now, consider either
+
+### Why are my changes not updated in my package set?
+
+Package sets are cached based on a git reference (e.g. tag or branch)
```
Just to be clear, this only happens if you're using a branch, or something like master, right? The typical usage should be with _tags_, where this shouldn't happen. Maybe we should spell that out.

mschristiansen replied:
You _can_ delete tags. Not good practice, but it happens and would create the same problem.

mschristiansen replied:
Yes, I was using a branch, while figuring out how to create a package set and then tagged it once things worked.

paf31 commented:
Thanks very much!
